third_party/amd/include/Dialect/TritonAMDGPU/IR/TritonAMDGPUOps.td

```diff
@@ -229,6 +231,43 @@ def BufferLoadOp : TT_AMDGPU_Op<"buffer_load", [
     }];
 }
 
+//===----------------------------------------------------------------------===//
+// BufferLoadToLocalOp
+//===----------------------------------------------------------------------===//
+
+def BufferLoadToLocalOp : TT_AMDGPU_Op<"buffer_load_to_local", [
+  AttrSizedOperandSegments,
+  MemoryEffects<[MemRead<GlobalMemory>]>,
+  MemoryEffects<[MemWrite<LocalMemory>]>,
+  TypesMatchWith<"result element type matches the pointed type of ptr", "result", "ptr", "getPointerTypeToElement($_self)">,
+  TypesMatchWith<"infer mask shape from offsets",
+                 "offsets", "mask", "getI1SameShape($_self)",
+                 "(cast<BufferLoadToLocalOp>($_op).getMask() == nullptr) || std::equal_to<>()">,
+]>{
+    let summary = "Load from a scalar base pointer and a tensor offset to local memory";
+    let description = [{
+      AMD Buffer load operation. Same as BufferLoad (see above) but writes to LDS instead of VGPR. }];
+    let arguments = (ins
+      TT_Ptr:$ptr,
+      I32Tensor:$offsets,
+      TTG_MemDescType:$result,
+      Optional<I32>:$stride,
+      DefaultValuedAttr<TT_CacheModifierAttr, "::mlir::triton::CacheModifier::NONE">:$cache,
+      Optional<TT_BoolTensor>:$mask,
+      Optional<TT_Tensor>:$other
```

Comment:
Can we have a better order for these fields: `$dest`, `$ptr`, `$offsets`, `$mask`, `$other`, `$stride`, `$cache`.

(We need to fix other op definitions to follow the above order to be consistent but that should be in another pull request.)